### PR TITLE
Refactor reports API and add unit tests

### PR DESCRIPTION
**Pull request by manitgupta**
Refactors reports generated APIs into their own reports.go file, and adds a UT for each of the APIs.

### CODE REVIEW

File: webv2/api/reports_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	// do nothing since we don't want to test report generation here, only the API.
+}
+
+func (r *GenerateReportMock) GenerateStructuredReport(driverName string, dbName string, conv *internal.Conv, badWrites map[string]int64, printTableReports bool, printUnexpecteds bool) reports.StructuredReport {
+	return reports.StructuredReport{
+		MigrationType: "test",
+	}
+}
+
+func TestGetReportFile(t *testing.T) {
```
Comment (asthamohta) [inline webv2/api/reports_test.go:35]: It looks like error flow is not tested, can we add tests for that?

File: webv2/api/rules_test.go [hunk 2]
```diff
@@ -18,7 +18,6 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-
```
Comment (asthamohta) [inline webv2/api/rules_test.go:21]: nit: we should remove this?

Comment (manitgupta) [reply webv2/api/rules_test.go:21]: Didn't get you, what should we remove?

Comment (asthamohta) [reply webv2/api/rules_test.go:21]: You've accidentally removed a line here and the whole file is now added to CL

**Review by asthamohta — APPROVED** [review]
LGTM

File: webv2/api/rules_test.go (hunk 2 above)
Comment (manitgupta) [reply webv2/api/rules_test.go:21]: Ahh gotcha thanks, will revert the change

**Pull request merged**